playbooks/roles/vhost/tasks/main.yml

```diff
@@ -39,6 +48,18 @@
     group: root
     mode: 0644
   when: COMMON_OBJECT_STORE_LOG_SYNC
+        and (ansible_distribution_release == 'precise' or ansible_distribution_release == 'trusty')
+
+# With systemd, log rotation should be run with an `ExecStop` directive
+# of `syslog.service` to ensure that rsyslogd will not be killed before attempt
+# of syncing logs.
+- name: Force logrotate on syslog stop
+  lineinfile:
+    path: /etc/systemd/system/syslog.service
+    line: "ExecStop={{ COMMON_OBJECT_STORE_LOG_SYNC_ON_EXIT }}"
+    insertafter: "^ExecStart=.*$"
+  when: COMMON_OBJECT_STORE_LOG_SYNC
+        and not (ansible_distribution_release == 'precise' or ansible_distribution_release == 'trusty')
```

Comment:
@Agrendalath, nit: it might be more readable if you expand the negation to each sub-condition